### ISSUE TITLE
bug: cannot find action for key apigateway.PutRestApi or apigateway.PutRestApi

### ISSUE THREAD

**Issue by davecardwell** [localstack/localstack#7330]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Running `apigateway put-rest-api` via `awslocal` works fine:
```sh
awslocal apigateway put-rest-api --body 'file:///tmp/test.json' --rest-api-id af086iav2w
```

However, when using IAM credentials it results in an exception:
```sh
AWS_SHARED_CREDENTIALS_FILE=/mnt/aws/credentials.admin aws --endpoint http://localstack:4566 --region us-east-1 apigateway put-rest-api --body 'file:///tmp/test.json' --rest-api-id af086iav2w
```
```
An error occurred (InternalError) when calling the PutRestApi operation (reached max retries: 4): exception while calling apigateway.PutRestApi: Traceback (most recent call last):
  File "/opt/code/localstack/localstack/aws/chain.py", line 90, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack/aws/chain.py", line 212, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/.venv/lib/python3.10/site-packages/localstack_ext/services/iam/policy_engine/handler.py.enc", line 14, in __call__
  File "/opt/code/localstack/.venv/lib/python3.10/site-packages/localstack_ext/services/iam/policy_engine/engine.py.enc", line 178, in is_request_allowed
  File "/opt/code/localstack/.venv/lib/python3.10/site-packages/localstack_ext/services/iam/policy_engine/engine.py.enc", line 79, in get_required_iam_actions_for_request
  File "/opt/code/localstack/.venv/lib/python3.10/site-packages/localstack_ext/services/iam/policy_engine/engine.py.enc", line 41, in get_actions_for_operation
localstack_ext.services.iam.policy_engine.engine.PolicyEvaluationException: Error in mapping between request and action map, cannot find action for key apigateway.PutRestApi or apigateway.PutRestApi
```

The same is true for `import-rest-api` (_“…cannot find action for key apigateway.ImportRestApi or apigateway.ImportRestApi”_).

> 

### Expected Behavior

I expect the command to check the IAM permissions and then to succeed, as it does via `awslocal`.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    # using https://github.com/localstack/localstack/blob/master/docker-compose-pro.yml with `ENFORCE_IAM: 1` set
    DEBUG=true LOCALSTACK_API_KEY={redacted} docker compose up -d

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    AWS_SHARED_CREDENTIALS_FILE=/mnt/aws/credentials.admin aws --endpoint http://localstack:4566 --region us-east-1 apigateway create-rest-api --name 'Test'

    # with the `id` from the previous command
    AWS_SHARED_CREDENTIALS_FILE=/mnt/aws/credentials.admin aws --endpoint http://localstack:4566 --region us-east-1 apigateway put-rest-api --body 'file:///tmp/test.json' --rest-api-id af086iav2w

### Environment

```markdown
- OS: macOS
- LocalStack: latest (1.3.1.dev, tag: latest, id: 7eebec566895, 📆 2022-12-13T14:02:55)
```

### Anything else?

Contents of /tmp/test.json in my examples above (via [AWS docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/create-api-using-swagger.html)):

```json
{
  "swagger": "2.0",
  "info": {
    "title": "Simple PetStore (OpenAPI)"
  },
  "schemes": [
    "https"
  ],
  "paths": {
    "/pets": {
      "get": {
        "responses": {
          "200": {
            "description": "200 response"
          }
        },
        "x-amazon-apigateway-integration": {
          "responses": {
            "default": {
              "statusCode": "200"
            }
          },
          "uri": "http://petstore-demo-endpoint.execute-api.com/petstore/pets",
          "passthroughBehavior": "when_no_match",
          "httpMethod": "GET",
          "type": "http"
        }
      }
    },
    "/pets/{petId}": {
      "get": {
        "parameters": [
          {
            "name": "petId",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response"
          }
        },
        "x-amazon-apigateway-integration": {
          "responses": {
            "default": {
              "statusCode": "200"
            }
          },
          "requestParameters": {
            "integration.request.path.id": "method.request.path.petId"
          },
          "uri": "http://petstore-demo-endpoint.execute-api.com/petstore/pets/{id}",
          "passthroughBehavior": "when_no_match",
          "httpMethod": "GET",
          "type": "http"
        }
      }
    }
  }
}
```

**Comment by calvernaz** [comment]
Hi @davecardwell, thanks for reporting this issue 👍 I did try to reproduce the example and I couldn't, however, I'm using the latest version.

Could you try using the latest version, and see if the error still happens?

**Comment by calvernaz** [comment]
Blocked by https://github.com/localstack/localstack-ext/pull/1239

**Comment by calvernaz** [comment]
@davecardwell seems I can reproduce because I'm using root credentials. The issue exists and there is a fix in progress for it. I'll ping you back once it lands upstream.

**Comment by dfangl** [comment]
Hi @davecardwell !
The latest image pushed just now should have the necessary fixes.
Just tested it, and it should now extract the correct action.
Thank you for reporting this issue! Please do not hesitate to reopen or open a new issue, if this issue is not resolved for you.
